protocols/virtualdefund/virtualdefund.go

```diff
@@ -203,8 +214,28 @@ func (o Objective) Crank(secretKey *[]byte) (protocols.Objective, protocols.Side
 		return &updated, sideEffects, WaitingForCompleteFinal, nil
 	}
 
-	// TODO: Implement ledger funding in https://github.com/statechannels/go-nitro/issues/480
-	return &updated, sideEffects, WaitingForCompleteLedgerDefunding, nil
+	if !updated.isAlice() && !updated.isLeftDefunded() {
+		ledgerSideEffects, err := updated.defundLedger(updated.ToMyLeft, secretKey)
+		if err != nil {
+			return &o, protocols.SideEffects{}, WaitingForNothing, fmt.Errorf("error updating ledger funding: %w", err)
+		}
+		sideEffects.Merge(ledgerSideEffects)
+	}
+
+	if !updated.isBob() && !updated.isRightDefunded() {
+		ledgerSideEffects, err := updated.defundLedger(updated.ToMyRight, secretKey)
+		if err != nil {
+			return &o, protocols.SideEffects{}, WaitingForNothing, fmt.Errorf("error updating ledger funding: %w", err)
+		}
+		sideEffects.Merge(ledgerSideEffects)
+	}
+
+	if fullyDefunded := updated.isLeftDefunded() && updated.isRightDefunded(); !fullyDefunded {
+		return &updated, sideEffects, WaitingForCompleteLedgerDefunding, nil
+	} else {
+		return &updated, sideEffects, WaitingForNothing, nil
+	}
```

Comment:
I think you don't need the `else` block here.

Reply:
Removed in https://github.com/statechannels/go-nitro/pull/540/commits/1e7a20f0d30ba31c84a43da41854c91156d2e381